sormas-ui/src/main/java/de/symeda/sormas/ui/AboutView.java

```diff
@@ -118,91 +157,114 @@ public AboutView() {
-						IOUtils.copy(preGeneratedDocumentStream, out);
-					}
-
-				}, (e) -> {
-				}), createFileNameWithCurrentDate(ExportEntityName.DATA_DICTIONARY, ".xlsx"))).extend(dataDictionaryButton);
-			}
+		if (shouldShowDataDictionaryLink()) {
+			Button dataDictionaryButton =
+				ButtonHelper.createButton(Captions.aboutDataDictionary, null, ValoTheme.BUTTON_LINK, CssStyles.BUTTON_COMPACT);
+			documentsLayout.addComponent(dataDictionaryButton);
 
-			// This link is hidden until an updated version of the document is provided
-			/*
-			 * Link technicalManualLink = new Link(
-			 * I18nProperties.getCaption(Captions.aboutTechnicalManual),
-			 * new ExternalResource(
-			 * "https://github.com/hzi-braunschweig/SORMAS-Project/files/2585973/SORMAS_Technical_Manual_Webversion_20180911.pdf"));
-			 * technicalManualLink.setTargetName("_blank");
-			 * documentsLayout.addComponent(technicalManualLink);
-			 */
+			new FileDownloader(new StreamResource(() -> new DownloadUtil.DelayedInputStream((out) -> {
+				try {
+					String documentPath = FacadeProvider.getInfoFacae().generateDataDictionary();
+					IOUtils.copy(Files.newInputStream(new File(documentPath).toPath()), out);
+				} catch (IOException e) {
+					LoggerFactory.getLogger(AboutView.class).error("Failed to generate data dictionary", e);
+
+					// fall back to pre-generated document
+					InputStream preGeneratedDocumentStream = new ClassResource("/doc/SORMAS_Data_Dictionary.xlsx").getStream().getStream();
+					IOUtils.copy(preGeneratedDocumentStream, out);
+				}
+
+			}, (e) -> {
+			}), createFileNameWithCurrentDate(ExportEntityName.DATA_DICTIONARY, ".xlsx"))).extend(dataDictionaryButton);
 		}
 
-		setSizeFull();
-		setStyleName("about-view");
-		addComponent(aboutContent);
-		setComponentAlignment(aboutContent, Alignment.MIDDLE_CENTER);
+		// This link is hidden until an updated version of the document is provided
+		/*
+		 * Link technicalManualLink = new Link(
+		 * I18nProperties.getCaption(Captions.aboutTechnicalManual),
+		 * new ExternalResource(
+		 * "https://github.com/hzi-braunschweig/SORMAS-Project/files/2585973/SORMAS_Technical_Manual_Webversion_20180911.pdf"));
+		 * technicalManualLink.setTargetName("_blank");
+		 * documentsLayout.addComponent(technicalManualLink);
+		 */
+
+		return documentsLayout;
 	}
 
-	@Override
-	public void enter(ViewChangeEvent event) {
+	private VerticalLayout createCustomHtmlSection(String caption, String fileName) {
+
+		VerticalLayout layout = new VerticalLayout();
+		layout.setSpacing(false);
+		layout.setMargin(new MarginInfo(true, false, false, false));
+
+		// Header
+		Label headerLabel = new Label(caption, ContentMode.HTML);
+		headerLabel.addStyleName(CssStyles.H1);
+		layout.addComponent(headerLabel);
+
+		// Label, loaded from custom file
+		Label htmlContent = new Label();
+		htmlContent.setContentMode(ContentMode.HTML);
+
+		Path customHtmlDirectory = Paths.get(FacadeProvider.getConfigFacade().getCustomFilesPath());
```

Comment:
Load the HTML content before setting up the layout and labels, and then make the equals check from below directly afterwards; that will allow you to return null if the files are empty without having to set up the objects